shopelectro/views/catalog.py

```diff
@@ -38,19 +39,27 @@ class CategoryTree(catalog.CategoryTree):
     category_model = models.Category
 
 
-def prepare_tile_products(products):
-    images = Image.objects.get_main_images_by_pages(
-        models.ProductPage.objects.filter(
-            shopelectro_product__in=products
+class ProductsAdapter:
```

Comment:
What does the `Adapter` mean here? It seems like nothing, because we may say, that every piece of code adapts something for something else. So let's call it more verbose like `TileProducts` (Tile contains product's name, image and category). Decide yourself

Reply:
@artemiy312 , it's not about tile products, but about converting one QuerySet to another for template format. We can move every queryset from such classes instead of using huge `prepare` inner functions in views.
I'll suggest different approach here